cloudimg/common.py

```diff
@@ -56,7 +57,13 @@ def _default_snapshot_name(self):
 
     @property
     def object_name(self):
-        return os.path.basename(self.image_path)
+        if not self._object_name:
```

Comment:
When I suggested before that the object name should be fixed to not contain trailing .xz, I meant right here.  Like making the property return this:

```suggestion
        return os.path.basename(self.image_path).rstrip('.xz')
```

This would mean right from the moment it's created, PublishingMetadata.object_name accurately gives you the object's intended name in S3 or Azure, which is almost surely how the code was intended to work and how it worked before any of the .xz decompression stuff was introduced.

This makes more sense than first initializing the object with an inaccurate name, and then in one code path during upload, patching it to the accurate name.